proto-wine/proton.cpp

```diff
@@ -0,0 +1,64 @@
+/* Copyright (c) 2019 Stanislaw Halik <sthalik@misaki.pl>
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ */
+
+#ifndef OTR_WINE_NO_WRAPPER
+
+#include <QtGlobal>
+#include <QString>
+#include <QProcessEnvironment>
+
+QProcessEnvironment make_steam_environ(const QString& proton_version)
+{
+    auto ret = QProcessEnvironment::systemEnvironment();
+    QString home = qgetenv("HOME");
+
+    auto expand = [&](QString x) {
+        x.replace("HOME", home);
+        x.replace("PROTON", proton_version);
+        return x;
+    };
+
+    QString path = expand(
+        "HOME/.local/share/Steam/steamapps/common/Proton PROTON/dist/bin:"
+        "HOME/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/bin:"
+        "HOME/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/bin"
+    );
+    path += ':'; path += qgetenv("PATH");
+    ret.insert("PATH", path);
+
+    QString library_path = expand(
+        ":HOME/.local/share/Steam/steamapps/common/Proton PROTON/dist/lib"
```

Comment:
Some consistency nitpicking: The colons here are at the beginning of each line whereas the colons are at the end for the path string elements.